### PR TITLE
DEPR: to_native_types, set_value, iteritems, union_many, to_perioddelta

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
IIUC the docbuild problem is 

```
/home/runner/work/pandas/pandas/doc/source/whatsnew/v2.0.0.rst:182: WARNING: Inline interpreted text or phrase reference start-string without end-string.
```

but line 182 in that file a) is not touched by this PR and b) looks fine to me.

File: doc/source/whatsnew/v2.0.0.rst [hunk 1]
```diff
@@ -164,6 +164,13 @@ Removal of prior version deprecations/changes
 - Removed deprecated :meth:`Index.is_type_compatible` (:issue:`42113`)
 - Removed deprecated :meth:`Index.is_mixed`, check ``index.inferred_type`` directly instead (:issue:`32922`)
 - Removed deprecated :meth:`Index.asi8` (:issue:`37877`)
+- Removed deprecated :meth:`DataFrame._AXIS_NUMBERS`, :meth:`DataFrame._AXIS_NAMES`, :meth:`Series._AXIS_NUMBERS`, :meth:`Series._AXIS_NAMES` (:issue:`33637)
```
Comment (phofl) [inline doc/source/whatsnew/v2.0.0.rst:167]: A tick is missing after the issue in this line

Comment (jbrockmendel) [reply doc/source/whatsnew/v2.0.0.rst]: thanks, fixed

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**